### ISSUE TITLE
Tridactyl keeps losing all of its settings.

### ISSUE THREAD

**Issue by notJerl** [tridactyl/tridactyl#1792]
Tridactyl version: 1.16.2pre2999-95db0a8
Firefox version: Mozilla Firefox 68.0.1
Operating system: win

*   Brief description of the problem:
Tridactyl randomly loses all of its settings.  This is *usually* temporary - unless it manages to save them to tridactylrc or sync them, depending on your `storageloc` setting.

*   Steps to reproduce:
Just use Tridactyl normally.  I haven't found a pattern to it.  It's happened immediately after browser restarts, but it's also just randomly happened on new tabs, and I've even come back to my computer to find it had happened on a tab that had been open the entire time.  If I restart Firefox before it saves the settings permanently, it'll come back with the settings intact, but if I'm too slow, they end up being lost.

*   Config (in a new tab, run `:viewconfig`, copy the url and paste it somewhere like gist.github.com):
https://pastebin.com/bscEujLR

**Comment by arcnmx** [comment]
If you've **ever** set `storageloc=local` on a machine, it will get stuck and load settings from there at browser start, and use local storage when you use `:set`, but then any sync events that happen to come in from then on will wipe everything. Then once that happens `:set` won't save anything properly. The setting is broken, don't use it. You can fix it by killing local storage on all affected machines:

```
:set storageloc sync
:sanitise tridactyllocal
```

And don't expect settings to work without having them sync'd/identical across all machines (or otherwise disabling firefox sync/account stuff).

**Comment by notJerl** [comment]
I had never set `storageloc=local` until *after* this started happening, and I only did it *because* this started happening.  Like I said, this sometimes happens at browser start, but it sometimes happens when just making a new tab.  Or like right now, when once again I came home from some errands to find it had happened on a Firefox install that has literally been sitting here doing nothing for about 24 hours.
